di-ipv-evcs-stub/lambdas/src/services/evcsService.ts

```diff
@@ -0,0 +1,158 @@
+
+import { config } from "../common/config";
+import { getSsmParameter } from "../common/ssmParameter";
+import { v4 as uuid } from "uuid";
+import PatchRequest from "../domain/patchRequest";
+
+const dynamoClient = config.isLocalDev
+  ? new DynamoDB({
+      endpoint: config.localDynamoDbEndpoint,
+      region: config.region,
+    })
+  : new DynamoDB({ region: config.region });
+
+export async function processPostUserVCsRequest(
+  userId: string,
+  postRequest: PostRequest,
+): Promise<ServiceResponse> {
+  console.info(`Post user record.`);
+  for (const persistVC of postRequest.persistVCs) {
+    const vcItem: EvcsVcItem = {
+      userId: userId,
+      vc: persistVC.vc,
+      vcSignature: persistVC.vc.split(".")[2],
+      state: persistVC.state,
+      metadata: persistVC.metadata!,
+      provenance: persistVC.provenance!,
+      ttl: await getTtl(),
+    };
+    await saveUserVC(vcItem);
+  }
+
+  return {
+    response: {
+      messageId: uuid(),
+    },
+    statusCode: 202,
+  };
+}
+
+export async function processGetUserVCsRequest(
+  userId: string,
+): Promise<ServiceResponse> {
+  console.info(`Get user record.`);
+  const getItemInput: QueryInput = {
+    TableName: config.evcsStubUserVCsTableName,
+    KeyConditionExpression: "#userId = :userIdValue",
+    FilterExpression: "#state = :stateValue",
+    ExpressionAttributeNames: {
+      "#userId": "userId",
+      "#state": "state",
+    },
+    ExpressionAttributeValues: {
+      ":userIdValue": { S: userId },
+      ":stateValue": { S: VcState.CURRENT },
+    },
+  };
+  // console.info(`Query Input - ${JSON.stringify(getItemInput)}`);
+  const items = (await dynamoClient.query(getItemInput)).Items ?? [];
+  console.info(`Total VCs retrived - ${items?.length}`);
+  const vcItems = items
+    .map((item) => unmarshall(item) as EvcsVcItem)
+    .map((evcsItem) => ({
+      vc: evcsItem.vc,
+      state: evcsItem.state as VcState,
+      metadata: evcsItem.metadata,
+    }));
+
+  return {
+    response: {
+      vcs: vcItems,
+      afterKey: "pagination will be implemented later",
+    },
+    statusCode: 200,
+  };
+}
+
+export async function processPatchUserVCsRequest(
+  userId: string,
+  postRequest: PatchRequest,
+): Promise<ServiceResponse> {
+  console.info(`Patch user record.`);
+  for (const updateVC of postRequest.updateVCs) {
+    const vcItem: EvcsVcItem = {
+      userId: userId,
+      vcSignature: updateVC.signature,
+      state: updateVC.state,
+      metadata: updateVC.metadata!,
+      ttl: await getTtl(),
+    };
+    await updateUserVC(vcItem);
+  }
+
+  return {
+    response: {
+      messageId: uuid(),
+    },
+    statusCode: 204,
+  };
+}
+
+async function saveUserVC(evcsVcItem: EvcsVcItem) {
+  console.info(`Save user vc.`);
+  const putItemInput: PutItemInput = {
+    TableName: config.evcsStubUserVCsTableName,
+    Item: marshall(evcsVcItem, {
+      removeUndefinedValues: true,
+    }),
+  };
+  await dynamoClient.putItem(putItemInput);
```

Comment:
Would it be useful to have error handling for these dynamo CRUD interactions? Not sure how likely it will fail but sounds like the kind of thing that would have pretty bad consequences for the service if we silently fail to save user's details

Reply:
At the very least I think we should be logging errors that we can trace / monitor

Reply:
EDIT: I can see you have a catch block in the handler, but it doesn't look like we are explicitly checking the status of the update / patch / post requests we make to the database to make sure we are getting a 200 successful response. 

Does this get handled automatically by the DynamoDb client? If so just ignore me 😅

Reply:
done now will be handled by try catch for Promise.all